test/functional/parser/heading_test.exs

```diff
@@ -14,6 +14,17 @@ defmodule Parser.HeadingTest do
     assert result == [%Block.Heading{content: "Heading", level: 1, lnb: 2}]
   end
 
+  test "Start heading option is respected" do
```

Comment:
You implemented it well, so functional tests do not seem very useful to me.

I would prefer some acceptance tests, but be aware that I will probably not merge this feature.